setup.py

```diff
@@ -56,11 +56,15 @@ def get_setup_version(reponame):
     import json, importlib
     basepath = os.path.split(__file__)[0]
     version_file_path = os.path.join(basepath, reponame, '.version')
-    version = importlib.import_module("version")
+    try:
+        from param import version
+    except:
+        embed_version(basepath)
+        version = importlib.import_module("version")
     if version is not None:
         return version.Version.setup_version(basepath, reponame, archive_commit="$Format:%h$")
     else:
-        print("WARNING: autover unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should refer to autover's documentation to bundle autover or add it as a dependency.")
+        print("WARNING: param>=1.6.0 unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should install param>=1.6.0.")
```

Comment:
I don't see how this warning will be reached. Either embed_version() worked and version's not None, or embed_version() fails...?

Reply:
Good point. Happy to remove it in another PR.

Reply:
I think that could be easily fixed by sticking in a try/except around embed version. i.e

```python
try:
   from param import version
except:
   version = embed_version(basepath) # Returns the module or None if failure
```

In this case, the warning is valid and it falls back to the JSON file. I'll make a PR for holoviews using this approach now...the key thing then will be to make sure this is used in other projects too (param doesn't need this but datashader probably does).

Reply:
I've made a PR to address this: #2532